Jenkinsfile

```diff
@@ -104,7 +104,72 @@ pipeline {
           }
         } // SNR plots
 
-        stage('Build and Test') {
+        stage ('Hardware Tests') {
+          agent {
+            label 'xcore.ai && uhubctl'
+          }
+          stages {
+            stage('Get repo') {
+              steps {
+                runningOn(env.NODE_NAME)
+                sh "mkdir ${REPO}"
```

Comment:
Minor, optional: you don't actually need this mkdir command - the dir() command will create a directory if it doesn't exist.